components/contentselector/model/contentmodel.cpp

```diff
@@ -512,7 +512,9 @@ void ContentSelectorModel::ContentModel::sortFiles()
             //dependencies appear.
             for (int j = i + 1; j < fileCount; j++)
             {
-                if (gamefiles.contains(mFiles.at(j)->fileName(), Qt::CaseInsensitive))
+                if (gamefiles.contains(mFiles.at(j)->fileName(), Qt::CaseInsensitive)
+                 || (!mFiles.at(i)->isGameFile() && gamefiles.isEmpty()
+                 && mFiles.at(j)->fileName().compare("Morrowind.esm", Qt::CaseInsensitive) == 0)) // Hack: implicit dependency on Morrowind.esm for dependency-less files
```

Comment:
This line looks odd, considering that Morrowind.esm can only be ever found at index 0.

Reply:
Uh... during the sorting it's not necessarily at index 0. Aren't the files added alphabetically in addFiles? And isn't the "point" of the bug that masterless plugins are ignored during the sorting and never get ordered after Morrowind.esm if they happen to be placed earlier than it in the order sortFiles() got?

Reply:
Hm, isn't that part of the problem? I haven't looked at the code recently, but the omwgame file (which is the type a dependency less esm should be mapped to) must always be at index 0. Maybe that isn't a problem, because it is sorted to the correct position later. But this surely looks odd.